src/test/lambda/local/detectLocalLambdas.test.ts

```diff
@@ -0,0 +1,152 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as del from 'del'
+import * as path from 'path'
+import { Uri, WorkspaceFolder } from 'vscode'
+import { detectLocalLambdas } from '../../../lambda/local/detectLocalLambdas'
+import { createTemporaryDirectory, saveTemplate } from './util'
+
+suite('detectLocalLambdas tests', () => {
+    test('detects no lambdas when workspaceFolders is undefined', async () => {
+        const actual = await detectLocalLambdas(undefined)
+
+        assert.ok(actual)
+        assert.equal(actual.length, 0)
+    })
+
+    test('detects no lambdas when workspaceFolders is empty', async () => {
+        const actual = await detectLocalLambdas([])
+
+        assert.ok(actual)
+        assert.equal(actual.length, 0)
+    })
+
+    test('detects no lambdas when template.y[a]ml does not exist', async () => {
+        const workspacePath = await createTemporaryDirectory('vsctk1')
```

Comment:
Since you have a reasonable amount of tests making then tearing down temp folders, consider moving that to a "before/after test" declaration. It keeps the test focused on the actual test, which is easier to read, and you don't have to make sure you properly set up/tear down within each test. You can see an example at the suite level [here](https://github.com/aws/aws-toolkit-vscode/blob/1320413f73096ba705fbf951848ece730e5894da/src/test/filesystem.test.ts#L18)

Reply:
`beforeEach`/`afterEach` are only available in BDD-style tests. I think that switching to BDD has many advantages, but it should be a separate PR.

Reply:
Update: Chris pointed out that they are available as `setup` and `teardown`. Updating.